rate/rate.py

```diff
@@ -0,0 +1,228 @@
+# coding: utf-8
+"""
+This is an XBlock designed to allow people to provide feedback on our
+course resources.
+"""
+
+import random
+
+import pkg_resources
+
+from xblock.core import XBlock
+from xblock.fields import Scope, Integer, String, List, Float
+from xblock.fragment import Fragment
+
+try:
+    from eventtracking import tracker
+except ImportError:
+    class tracker(object):  # pylint: disable=invalid-name
+        """
+        Define tracker if eventtracking cannot be imported. This is a workaround
+        so that the code works in both edx-platform and XBlock workbench (the latter
+        of which does not support event emission). This should be replaced with XBlock's
+        emit(), but at present, emit() is broken.
+        """
+        def __init__(self):
+            """ Do nothing """
+            pass
+
+        @staticmethod
+        def emit(param1, param2):
+            """ In workbench, do nothing for event emission """
+            pass
+
+
+class RateXBlock(XBlock):
+    """
+    This is an XBlock -- eventually, hopefully an aside -- which
+    allows you to rate content in the course. We've wanted this for a
+    long time, but Dartmouth finally encourage me to start to build
+    this.
+    """
+
+    default_prompt = {'freeform': "Please provide us feedback on this section.",
+                      'likert': "Please rate your overall experience with this section.",
+                      'mouseovers': ["Excellent", "Good", "Average", "Fair", "Poor"],
+                      'icons': [u"😁", u"😊", u"😐", u"😞", u"😭"]}
+
+    # This is a list of prompts. If we have multiple elements in the
+    # list, one will be chosen at random. This is currently not
+    # exposed in the UX. If the prompt is missing any portions, we
+    # will default to the ones in default_prompt.
+    prompts = List(
+        default=[{'freeform': "What could be improved to make this section more clear?",
+                  'likert': "Was this section clear or confusing?"}],
+        scope=Scope.settings,
+        help="Freeform user prompt",
+        xml_node=True
```

Comment:
Nominally, it means that the serialized representation will not stick things in attributes, but as a child node. 

    <rate>
       <prompts> </prompts>
    </rate>

rather than:

   <rate prompts="...">